Sources/Sugar/Authentication/UserType.swift

```diff
@@ -7,22 +7,36 @@ public protocol UserType {
     associatedtype Login: HasReadablePassword
     associatedtype Registration: HasReadablePassword
     associatedtype Update: Decodable
+    associatedtype Public: Content
 
     init(_: Registration) throws
+    func update(with: Update) throws
+
     static func logIn(with: Login, on worker: DatabaseConnectable) -> Future<Self>
-    func update(using: Update) throws
+    static func preRegister(with: Registration, on worker: DatabaseConnectable) throws -> Future<Void>
```

Comment:
Especially these 2 could use some API docs to indicate what it could be used for

Reply:
I agree and added. Let's add for everything else when the package is more 'ready'.